### ISSUE TITLE
MissingManifestResourceException: Could not find any resources appropriate for the specified culture or the neutral culture.

### ISSUE THREAD

**Issue by CrispyDrone** [Testura/Testura.Mutation#53]
Hi,

I'm trying out the console application on one of our projects at work. However, I'm encountering a `MissingManifestResourceException` while the baseline is being run.

When I run the tests through `dotnet test` the tests all pass.

The binary resource is embedded in a separate module but during its compilation it seems to be skipped if I'm correct:

```xml
<EmbeddedResource Update="Resources\Messages.resx">
  <Generator>PublicResXFileCodeGenerator</Generator>
  <LastGenOutput>Messages.Designer.cs</LastGenOutput>
</EmbeddedResource>
```

In the following method, the `Include` attribute is not present for this `resx` resource.

```csharp
        private IList<ResourceDescription> GetEmbeddedResources(string assemblyName, string projectPath)
        {
            var resources = new List<ResourceDescription>();
            var doc = XDocument.Load(projectPath);
            var embeddedResources = doc.Descendants().Where(d => d.Name.LocalName.Equals("EmbeddedResource", StringComparison.InvariantCultureIgnoreCase));
            foreach (var embeddedResource in embeddedResources)
            {
                var paths = embeddedResource.Attribute("Include")?.Value;
                if (paths == null)
                {
                    continue;
                }

                foreach (var path in paths.Split(';'))
                {
                    var pathFixed = path.Split('\\');
                    var resourcePath = Path.Combine(Path.GetDirectoryName(projectPath), path);

                    resources.Add(new ResourceDescription(
                        $"{assemblyName}.{string.Join(".", pathFixed)}",
                        () => File.OpenRead(resourcePath),
                        true));
                }
            }

            return resources;
        }
```

Which means it seems to be skipping this embedded resource while emitting the compilation into the baseline directory.

![image](https://user-images.githubusercontent.com/37639058/77152878-15698980-6a99-11ea-9aee-079083334d89.png)

**Comment by MilleBo** [comment]
Hey! 

Yeah that seems correct. I haven't worked with a lot of embedded resources so this seems like something I should fix. Do you have an example project that I can test with? 

Gonna try to set up something myself too.

**Comment by MilleBo** [comment]
So I found multiple problems with my solution. 

1. I didn't look for Update as you mention 
2. It didn't handle .resx file as I do a simple File.Open 

I have fixed both and will test with a couple of different project types first. But I will try to upload a new version soon.

**Comment by MilleBo** [comment]
I have upload a new release (1.7.0):

https://github.com/Testura/Testura.Mutation/releases/tag/v1.7.0-Console

Would love if you could verify that it work better. I tried multiple different projects and could read the embedded resource fine with this fix.

**Comment by CrispyDrone** [comment]
Hi,

Thanks for your response and you quick work!

I will be able to verify this at work tomorrow and will keep you updated.

**Comment by CrispyDrone** [comment]
Hi again,

I tested it with the application at work however it seems it's still not working. I had a look at the assembly that's generated by the baseline and it seems the name of the resource is different compared to the assembly that's generated when building with visual studio.

I blocked out the name of the organization. The one without the organization name is being generated by visual studio, while Testura seems to add a prefix to the namespace.

Don't judge me for this, but this application has inconsistent namespaces haha :)

![mismatched-namespace-question](https://user-images.githubusercontent.com/37639058/77346548-b4f77800-6d36-11ea-9960-8714d191f069.png)

So to summarize, I don't know if this is the root of the issue but the namespace in Testura's case seems to be different, as well as the "Length".

**Comment by MilleBo** [comment]
Interesting case ;-) 

Yeah I always add the assembly name when creating the resource like this:

`$"{assemblyName}.{string.Join(".", resourceName.Split('\\'))}",`

It feels like that should be the norm but guess I have to actually check if the resource contain the assembly name somehow. I will try to look into it tomorrow!

**Comment by MilleBo** [comment]
But just to be sure, I guess you use a default namespace for the whole application that doesn't match the assembly name? Or is there actually a way to just change the namespace for a resource?

**Comment by MilleBo** [comment]
Another day and another release:

https://github.com/Testura/Testura.Mutation/releases/tag/v1.7.1-Console

Now we will look at the "RootNamespace" attribute in the .csproj file instead of just using the assembly name. Hopefully this one will work better and I haven't missed another case.

**Comment by CrispyDrone** [comment]
Hello,

I tested it again and can confirm it works. Thanks a lot for your hard work! Now I'll be able to give Testura.Mutation a try :)

**Comment by MilleBo** [comment]
No problems and thanks for reporting the bug! :-)
